frontend/lib/src/features/account_details/account_details_page.dart

```diff
@@ -39,7 +39,9 @@ class AccountDetailsPage extends StatelessWidget {
           return const CircularProgressIndicator();
         }
 
-        spendingBreakdownController.getSpendingBreakdown();
+        spendingBreakdownController.getSpendingBreakdown(
+          accountId: arguments.accountId,
+        );
```

Comment:
since this is an asynchronous fn, this should be in another futurebuilder

it would be better if we combine it with the previous fn call but theyre from different controllers so just use 2 futurebuilder

1. then combine the returning of the progress indicator by checking both snapshots in one if block

2. or you can wrap only the Graph portion of the page with the future builder

Reply:
its actually better to have two separate if blocks checking for the connection state

Reply:
its actually better to have two separate if blocks checking for the connection state

Reply:
its actually better to have two separate if blocks checking for the connection state

Reply:
its actually better to have two separate if blocks checking for the connection state